main.cc

```diff
@@ -61,27 +48,42 @@ void buildBoard(std::vector<std::vector<Piece *>> &board) {
 }
 
 
-Observer constructPlayer(ChessGame *theGame, std::string player, std::vector<Piece> *pieceSet) {
+Observer * constructPlayer(ChessGame *theGame, std::string player, std::vector<Piece *> *pieceSet) {
+
+    /* 
+    This line is to be deleted once levels 1,2,3 are working. The only reason this line 
+    exists is b/c u can't have an used parameter in a function. It gives u compilation errors
+    */
+    if (((*pieceSet)[0])->getX()) {
+
+    };
+
     if (player == "human") {
         Human humanPlayer{theGame, "human"};
```

Comment:
The constructor for this will be changing because Human will now also contain a pieceArray field (to check for stalemating/checkmating).